HiP-EventStoreLib/EventStoreLlp/EventDataExtensions.cs

```diff
@@ -20,11 +23,24 @@ public static EventData ToEventData(this IEvent ev, Guid eventId, IDictionary<st
 
             var eventHeaders = new Dictionary<string, object>(headers ?? NoHeaders)
             {
-                { EventClrTypeHeader, type.AssemblyQualifiedName }
+                { EventClrTypeHeader, type.FullName }
             };
-            
-            var data = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(ev));
-            var metadata = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(eventHeaders));
+
+            if (typeof(ICustomEvent).IsAssignableFrom(type))
+            {
+                var customEvent = (ICustomEvent)ev;
+                eventHeaders = eventHeaders.Union(customEvent.GetAdditionalMetadata())
```

Comment:
`Union` won't prevent you from duplicate keys, if that's what you tried to use it for. Thus, you could also just use `Concat` which is more efficient - it doesn't make a huge difference, I just wanted to let you know.